p2pserver/link/link.go

```diff
@@ -110,30 +110,32 @@ func (this *Link) GetRXTime() time.Time {
 }
 
 func (this *Link) Rx() {
-	reader := bufio.NewReaderSize(this.conn, common.MAX_BUF_LEN)
+	if this.conn != nil {
+		reader := bufio.NewReaderSize(this.conn, common.MAX_BUF_LEN)
```

Comment:
save `this.conn` to a local variable first to avoid data race problem? and return directly if conn==nil to avoid another calling of `disconnectNotify`

Reply:
code updated, please check